dotCMS/src/main/java/com/dotmarketing/webdav/BasicFolderResourceImpl.java

```diff
@@ -58,36 +60,32 @@ public Resource createNew(String newName, InputStream in, Long length, String co
         if(!path.endsWith("/")){
             path = path + "/";
         }
-        if(!dotDavHelper.isTempResource(newName)){
 
+        if(!dotDavHelper.isTempResource(newName)){
             try {
-            	IFileAsset f = null;
             	dotDavHelper.setResourceContent(path + newName, in, contentType, null, java.util.Calendar.getInstance().getTime(), user, isAutoPub);
-                f = dotDavHelper.loadFile(path + newName,user);
-                FileResourceImpl fr = new FileResourceImpl(f, f.getFileName());
-                return fr;
+                final IFileAsset iFileAsset = dotDavHelper.loadFile(path + newName,user);
+                final Resource fileResource = new FileResourceImpl(iFileAsset, iFileAsset.getFileName());
+                return fileResource;
                 
-            }catch (DotSecurityException dotE){
+            }catch (Exception e){
             	Logger.error(this, "An error occurred while creating new file: " + (newName != null ? newName : "Unknown") 
                 		+ " in this path: " + (path != null ? path : "Unknown") + " " 
-                		+ dotE.getMessage(), dotE);
-            	throw new DotRuntimeException(dotE.getMessage(), dotE);
-            	
-            }catch (Exception e) {
-                Logger.error(this, "An error occurred while creating new file: " + (newName != null ? newName : "Unknown") 
-                		+ " in this path: " + (path != null ? path : "Unknown") + " " 
                 		+ e.getMessage(), e);
+            	throw new DotRuntimeException(e.getMessage(), e);
+            }
+        } else {
+            try {
+                originalPath = (!originalPath.endsWith("/"))?originalPath + "/":originalPath;
+                final File tempFile = dotDavHelper.createTempFile("/" + host.getHostname() + originalPath + newName);
+                FileUtils.copyStreamToFile(tempFile, in, null);
+                final Resource tempFileResource = new TempFileResourceImpl(tempFile, originalPath + newName, isAutoPub);
+                return tempFileResource;
+            } catch (Exception e){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)